packages/e2e-tests/fixtures/blocks/core__query__deprecated-1.serialized.html

```diff
@@ -0,0 +1,3 @@
+<!-- wp:query {"query":{"perPage":null,"pages":0,"offset":0,"postType":"post","categoryIds":[],"tagIds":[],"order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":true},"layout":{"type":"list"}} -->
```

Comment:
As long as default value for attributes is used, it should look rather like:

```html
<!-- wp:query -->
 <div class="wp-block-query"></div>
 <!-- /wp:query -->
```

Reply:
I'll investigate. I'm wondering if it has to do with these properties being `objects` that is not so common in core blocks. For example `queryId` is not here 🤔

Reply:
I wouldn't be surprised. I think you know already my opinion about using complex key/value-based objects as attributes. I would avoid them in the first place, so if that creates an issue, we could use the opportunity that deprecation is inevitable to refactor that as well. Anyway, I hope that is something else and you don't have to end up with a nuclear option 🤣

Reply:
In some cases (Query being the main case that comes in my mind) it makes sense to use `objects` and we could always explore adding better support for them. Anyway..

How do you feel about this:https://github.com/WordPress/gutenberg/pull/30804/commits/120c828ebaf1466e4dabdb39f15b9c1660476b0a

We could always use `isEqual` but felt like we could avoid the extra call for most cases that is not needed (not type `object`.

@youknowriad, @gziolo , @mcsf